ecs-cli/modules/cli/compose/entity/service/service.go

```diff
@@ -121,6 +123,9 @@ func (s *Service) LoadContext() error {
 		if targetGroupArn != "" && loadBalancerName != "" {
 			return errors.Errorf("[--%s] and [--%s] flags cannot both be specified", flags.LoadBalancerNameFlag, flags.TargetGroupArnFlag)
 		}
+		if len(loadBalancerList) != 0 {
```

Comment:
You need to specify role flag to allow Amazon ECS to make calls to your load balancer or target group on your behalf. It seems like if you do the validation like this users cannot do that :P